tests/testsCore/src/androidTest/kotlin/com/wire/android/tests/core/pages/ConversationViewPage.kt

```diff
@@ -0,0 +1,345 @@
+
+            // Perform fling (fast scroll) to the bottom
+            val success = scrollable.flingToEnd(10)
+            println("✅ Scrolled to bottom: $success")
+        } catch (e: Exception) {
+            println("❌ Failed to scroll: ${e.message}")
+        }
+    }
+
+    fun tapDownloadButtonOnVideoFile(): ConversationViewPage {
+        UiWaitUtils.waitElement(downloadButtonOnVideoFile).click()
+        return this
+    }
+
+    fun tapToPlayVideoFile(): ConversationViewPage {
+        UiWaitUtils.waitElement(videoDurationLocator).click()
+        return this
+    }
+
+    fun assertWireAppIsNotInForeground() {
+        val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        val currentPackage = device.currentPackageName
+        assertFalse(
+            "❌ Wire app is still in foreground: $currentPackage",
+            currentPackage.contains("APP_")
+        )
+    }
+
+    fun typeMessageInInputField(message: String): ConversationViewPage {
+        val inputField = UiWaitUtils.waitElement(messageInputField)
+        inputField.click()
+        // Use shell command to input text
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("input text ${message.replace(" ", "%s")}")
+        return this
+    }
+
+    fun clickSendButton(): ConversationViewPage {
+        UiWaitUtils.waitElement(sendButton).click()
+        return this
+    }
+
+    fun assertMessageSentIsVisible(message: String): ConversationViewPage {
+        val messageSelector = UiSelectorParams(text = message)
+        val messageElement = UiWaitUtils.waitElement(messageSelector)
+
+        Assert.assertTrue(
+            "❌ Message '$message' is not visible in the conversation",
+            !messageElement.visibleBounds.isEmpty
+        )
+        return this
+    }
+
+    fun assertMessageNotVisible(text: String) {
+        val element = findElementOrNull(UiSelectorParams(text = text))
+
+        Assert.assertTrue(
+            "❌ Message '$text' is still visible on the screen.",
+            element == null || element.visibleBounds.isEmpty
+        )
+    }
+
+    fun assertMessageReceivedIsVisible(message: String): ConversationViewPage {
+        val messageSelector = UiSelectorParams(text = message)
+        val messageElement = UiWaitUtils.waitElement(messageSelector)
+
+        Assert.assertTrue(
+            "❌ Message '$message' is not visible in the conversation",
+            !messageElement.visibleBounds.isEmpty
+        )
+        return this
+    }
+
+    fun tapBackButtonOnConversationViewPage(): ConversationViewPage {
+        UiWaitUtils.waitElement(backButton).click()
+        return this
+    }
+
+    fun tapMessageInInputField(): ConversationViewPage {
+        val inputField = UiWaitUtils.waitElement(messageInputField)
+        inputField.click()
+        return this
+    }
+
+    fun tapSelfDestructTimerButton(): ConversationViewPage {
```

Comment:
I believe we should follow the same patterns for namings of the buttons as we have in the android code. Is the self deleting messages button there also called `SelfDestruct`? If so, we can leave the related functions like this. :) But if it's called `SelfDelete` in the android app code, we should use the same wording, just to avoid confusion.